frontend/src/metabase/visualizations/echarts/cartesian/option/axis.ts

```diff
@@ -0,0 +1,190 @@
+// eslint-disable-next-line no-restricted-imports
+import moment from "moment-timezone";
+import type { AxisBaseOptionCommon } from "echarts/types/src/coord/axisCommonTypes";
+import type { CartesianAxisOption } from "echarts/types/src/coord/cartesian/AxisModel";
+import type {
+  ComputedVisualizationSettings,
+  RemappingHydratedDatasetColumn,
+  RenderingContext,
+} from "metabase/visualizations/types";
+import type { CartesianChartModel } from "metabase/visualizations/echarts/cartesian/model/types";
+
+import { isNumeric } from "metabase-lib/types/utils/isa";
+
+const getAxisNameDefaultOption = (
+  { getColor, fontFamily }: RenderingContext,
+  nameGap: number,
+  name?: string,
+): Partial<AxisBaseOptionCommon> => ({
+  name,
+  nameGap,
+  nameLocation: "middle",
+  nameTextStyle: {
+    color: getColor("text-dark"),
+    fontSize: 14,
+    fontWeight: 900,
+    fontFamily,
+  },
+});
+
+const getTicksDefaultOption = ({ getColor, fontFamily }: RenderingContext) => {
+  return {
+    hideOverlap: true,
+    color: getColor("text-dark"),
+    fontSize: 12,
+    fontWeight: 900,
+    fontFamily,
+  };
+};
+
+export const getXAxisType = (settings: ComputedVisualizationSettings) => {
+  switch (settings["graph.x_axis.scale"]) {
+    case "timeseries":
+      return "time";
+    case "linear":
+      return "value";
+    default:
+      // TODO: implement histogram
+      return "category";
+  }
+};
+
+export const buildDimensionAxis = (
+  settings: ComputedVisualizationSettings,
+  column: RemappingHydratedDatasetColumn,
+  renderingContext: RenderingContext,
+): CartesianAxisOption => {
+  const { getColor } = renderingContext;
```

Comment:
Nit: we should either destructure both `getColor` and `formatValue` or neither, and if we do destructure either always do it in the parameters like the above functions, or always in the body.

I personally prefer not destructuring because it makes it explicit that this function is from the `renderingContext` parameter, and not imported from another file.